index.js

```diff
@@ -368,16 +371,24 @@ class Ky {
 			if (ms !== 0 && this._retryCount > 0) {
 				await delay(ms);
 
+				const beforeRetryResults = new Set();
+
 				for (const hook of this._options.hooks.beforeRetry) {
-					// eslint-disable-next-line no-await-in-loop
-					await hook(
-						this.request,
-						this._options,
-						error,
-						this._retryCount,
+					beforeRetryResults.add(
```

Comment:
It would be more consistent with the behavior of our other hooks if returning the symbol short-circuited the loop. Meaning, return immediately inside of the loop, preventing any further `beforeRetry` hooks from running.

Reply:
I wasn't sure about that, my reasoning was that you'd want to have all your hooks running before making the "cancel retry" decision.

Reply:
I get what you are saying. I do see how this case is a bit different, as the symbol is not a true "value" being returned by the hook, unlike the other cases where a hook may return a `Request` or `Response` instance.

Still, I think it would be surprising for `beforeRetry` to behave differently in this respect than the other hook types. IMO, we need to have a holistic approach on whether returning a value from a hook short-circuits the chain. And we should apply that logic consistently.